feature/add_assignee_and_requester_id (#261)
fivetran/dbt_zendesk: CHANGELOG.md

# dbt_zendesk v1.6.0

> [PR #261](https://github.com/fivetran/dbt_zendesk/pull/261) includes the following update:

Reviewer: Once the below suggested updates are applied.
```suggestion
[PR #261](https://github.com/fivetran/dbt_zendesk/pull/261) includes the following updates:
```

> ## Schema/Data Change

Reviewer: Once the below suggested updates are applied.
```suggestion
## Schema/Data Changes
```

> **1 total change • 1 possible breaking change**

Reviewer: Once the below updates are applied.

```suggestion
**3 total changes • 2 possible breaking changes**
```

| **Data Model** | **Change type** | **Old** | **New** | **Notes** |
| -------------- | --------------- | ------- | ------- | --------- |

> | [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) **(Breaking)** | New columns | | `assignee_id`, `requester_id` | You will need to run a full refresh to backfill historical values. |

Reviewer: I think this actually needs to be separated out into two rows. 
- `assignee_id` is brought in by default now, and is a breaking change. 
- `requester_id` can be brought in, but not by default, so it's a schema change, but not a breaking one.

> 

Reviewer: `int_zendesk__ticket_historical_status` is materialized as a table, so this is actually a breaking schema change for Quickstart customers, so I'd say we still need to update these column removals in the CHANGELOG.

Author: Ah, missed that it was materialized as table. Good callout! Will update.

# dbt_zendesk v1.6.0

[PR #261](https://github.com/fivetran/dbt_zendesk/pull/261) includes the following updates:

## Schema/Data Changes

> **3 total changes • 2 possible breaking changes**

Reviewer: I'm realizing they are all **possible** breaking changes. My bad. 
```suggestion
**3 total changes • 3 possible breaking changes**
```

| **Data Model** | **Change type** | **Old** | **New** | **Notes** |
| -------------- | --------------- | ------- | ------- | --------- |
| [`int_zendesk__ticket_historical_status`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.int_zendesk__ticket_historical_status) **(Breaking)** | Removed fields | `ticket_status_counter`, `unique_status_counter` | | Columns were deprecated in v1.5.0 and are now removed. |

> | [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) **(Breaking)** | New field | | `assignee_id` | Added by default when `assignee_id` is in `ticket_field_history_columns`. You will need to run a full refresh to backfill historical values. |

Reviewer: Let's switch lines 12/13 with 11--end model changes should still go first.

> | [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) **(Breaking)** | New field | | `requester_id` | Added when `requester_id` is included in the `ticket_field_history_columns` variable. You will need to run a full refresh to backfill historical values.  |

Reviewer 2: ```suggestion
| [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) | New field | | `assignee_id` | Added by default when `assignee_id` is in `ticket_field_history_columns`. You will need to run a full refresh to backfill historical values. |
| [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) | New field | | `requester_id` | Added when `requester_id` is included in the `ticket_field_history_columns` variable. You will need to run a full refresh to backfill historical values.  |
```

Also what do you think about the following:
1. Removing the full refresh note (since they're automatically run for Quickstart breaking changes) or at least specifying that it's relevant to dbt Core only
2. Mentioning that previously only the `assignee/requester_name` field would be included and that now it's both id + name
3. Referring to how the `ticket_field_history_columns` variable looks like the Quickstart UI (**Ticket Field History Columns**) and either including the dbt Core name or linking to its README [section](https://github.com/fivetran/dbt_zendesk/blob/feature/add_assignee_and_requester_id/README.md#tracking-ticket-field-history-columns)

Author: 1. I think I want to leave in the full refresh but will change the wording to "dbt core users will need to run a full refresh..." 2. Agree, will do. 3. Would this be a separate bullet or a note within the schema change table? Not totally sure how to format this, and also want to be cognizant of not making the changelog too long.

Author: @fivetran-jamie the new fields are breaking, no?

Reviewer 2: 3. Could look like: "Added by default when `assignee_id` is included in the **Ticket Field History Columns** variable (`ticket_field_history_columns`)"  I know I'm getting pedantic here lol, up to you

And re:
> the new fields are breaking, no?

Yeah, but we just usually don't put "Breaking" in the Data Model column per se (this would go in Notes typically). You do also highlight that they're breaking on line 7

> | [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) | New field | | `requester_id` | Added when `requester_id` is included in the `ticket_field_history_columns` variable. You will need to run a full refresh to backfill historical values.  |

Reviewer: Yeah this is actually potentially breaking too. 
```suggestion
| [`zendesk__ticket_backlog`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.zendesk__ticket_backlog) **(Breaking)** | New field | | `requester_id` | Added when `requester_id` is included in the `ticket_field_history_columns` variable. You will need to run a full refresh to backfill historical values.  |
```

> | [`int_zendesk__ticket_historical_status`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.int_zendesk__ticket_historical_status) **(Breaking)** | Removed fields | `ticket_status_counter`, `unique_status_counter` | | Columns were deprecated in v1.5.0 and are now removed. |

Reviewer 2: ```suggestion
| [`int_zendesk__ticket_historical_status`](https://fivetran.github.io/dbt_zendesk/#!/model/model.zendesk.int_zendesk__ticket_historical_status) | Removed fields | `ticket_status_counter`, `unique_status_counter` | | Columns were deprecated in v1.5.0 and are now removed. |
```

Author: Field removal will be a breaking change for core users due to the materialization. Should I note that it's just for core users?

Reviewer 2: You do highlight that it could be breaking on line 7, but if you wanna do so here as well, I'd put it in the Notes rather than the Data Model column
